cloudnetwork/v1/types.go

```diff
@@ -0,0 +1,80 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CloudPrivateIPConfig performs assignments of a private IP address to the
+// primary NIC associated with cloud VMs. This is done by specifying the IP and
+// Kubernetes node which the IP should be assigned to. This CRD is intended to
+// be used by the network plugin which manages the cluster network. The spec
+// side represents the desired state requested by the network plugin, and the
+// status side represents the current state that this CRD's controller has
+// executed. No users will have permission to modify it, and if a cluster-admin
+// decides to edit it for some reason, their changes will be overwritten the
+// next time the network plugin reconciles the object. Note: the CR's name
+// specifies the requested private IP address and must match its `.spec.ip`
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=cloudprivateipconfig,scope=Cluster
+type CloudPrivateIPConfig struct {
```

Comment:
Naming question: does this need to actually be private? Or is it just _additional_?  I understand the underlying implementation currently only allows for private IPs, but how much of that is an implementation detail?

In other words, would it make sense for this to be be named CloudIPConfig? or just `IP.cloud.network.openshift.io`?

Reply:
I would like to keep it "private" (and I think this is also something we explicitly mentioned during our initial meetings in January) because there is a distinction between private and external IP addresses on cloud provider VMs. The control loop for this CRD will (which essentially only exists as to make egress IP happen in the cloud) should only assign private IP addresses. I think it makes sense to be explicit here and have that reflected in the naming convention.